src/policy/compiler.rs

```diff
@@ -148,14 +149,18 @@ struct CompilerExtData {
     /// (total length of all witness pushes, plus their own length prefixes)
     /// for fragments that can be dissatisfied without failing the script.
     dissat_cost: Option<f64>,
+    /// Signature Type for deciding the number of bytes required for [`CompilerExtData::sat_cost`]
+    /// and [`CompilerExtData::dissat_cost`] for the given script.
+    sig_type: PhantomData<Ctx>,
 }
 
-impl Property for CompilerExtData {
+impl<Ctx: ScriptContext> Property for CompilerExtData<Ctx> {
```

Comment:
Instead of introducing `Ctx` to `Property` and `CompilerExtData`, we should introduce it to `from_pk`, `from_pkh`, `from_multi`, and `from_multi_a` methods respectively. This should cleanup a lot of code here.

That is,
```diff
     /// Type property of the `PkK` fragment
-    fn from_pk_k() -> Self;
+    fn from_pk_k<Ctx: ScriptContext>() -> Self;
 
     /// Type property of the `PkH` fragment
-    fn from_pk_h() -> Self;
+    fn from_pk_h<Ctx: ScriptContext>() -> Self;
```

You would need to modify this for multi and multi_a too.

Reply:
Addressed in 2565f5e